### DISCUSSION TITLE
proxy not works with https sites

### DISCUSSION THREAD

**Discussion by denis-marecky** [nodejs/undici#662] [Q&A]
400 bad request when using proxy & trying to send https requests. http requests works fine

**Comment by ronag** [comment]
Please provide a reproducible example

**Comment by denis-marecky** [comment]
async function test() {
  const { Client } = require('undici')

  const serverUrl = `https://facebook.com:443`
  const proxyUrl = `http://122.155.165.191:3128`

  const client = new Client(proxyUrl)
  client.on('connect', () => {
    console.log(`Connected`)
  })
  client.on('disconnect', arg => {
    console.log(`Disconnected: ${arg}`)
  })

  const response = await client.request({
    method: 'GET',
    path: serverUrl,
  })

  console.log(`Status: ${response.statusCode}`)

  response.body.setEncoding('utf8')
  let data = ''
  for await (const chunk of response.body) {
    data += chunk
  }
  console.log(data)

  client.close()
}
test()

**Comment by mcollina** [comment]
Running that script I am getting:

```
(node:55050) UnhandledPromiseRejectionWarning: Error: connect ECONNREFUSED 122.155.165.191:3128
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:55050) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:55050) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Disconnected: [object Object]
```

It seems that host is down.

**Comment by denis-marecky** [comment]
It seems that proxy goes down. I took some public to not share my own private. Try any other public proxy currently alive. We test that on few servers

**Comment by mcollina** [comment]
I do not have a public proxy. I would recommend you to send a PR/build an example similar to https://github.com/nodejs/undici/blob/main/test/proxy.js.

**Comment by Ethan-Arrowood** [comment]
Moving to discussions as this seems like a help request
